matrix/Matrix.hpp

```diff
@@ -290,8 +290,8 @@ class Matrix
 
     void print() const
     {
-        char buf[1000];
-        write_string(buf, 1000);
+        char * buf = new char[15*N*M];
```

Comment:
Do you really want to spend 15 characters for every number? Seems excessive to me... but maybe you have other applications.